legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/depot/api/DepotMetadataApi.java

```diff
@@ -0,0 +1,142 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.sdlc.server.depot.api;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.message.BasicNameValuePair;
+import org.eclipse.collections.api.factory.Lists;
+import org.finos.legend.sdlc.domain.model.entity.Entity;
+import org.finos.legend.sdlc.server.depot.DepotConfiguration;
+import org.finos.legend.sdlc.server.depot.DepotServerInfo;
+import org.finos.legend.sdlc.server.depot.model.DepotProjectId;
+import org.finos.legend.sdlc.server.depot.model.DepotProjectVersion;
+import org.finos.legend.sdlc.server.error.LegendSDLCServerException;
+import org.finos.legend.sdlc.server.tools.StringTools;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DepotMetadataApi extends BaseDepotApi implements MetadataApi
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(DepotMetadataApi.class);
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final String GET_ENTITIES_PATH = "/api/projects/%s/%s/versions/%s";
+    private static final String GET_DEPENDENCIES_PATH = "/api/projects/%s/%s/versions/%s/projectDependencies";
+
+    @Inject
+    public DepotMetadataApi(DepotConfiguration configuration)
+    {
+        super(DepotServerInfo.newServerInfo(configuration.getServerConfiguration()), configuration.getAuthClientInjector());
+    }
+
+    @Override
+    public List<Entity> getEntities(String projectId, String versionId)
+    {
+        LegendSDLCServerException.validateNonNull(projectId, "Project id may be null");
+        LegendSDLCServerException.validateNonNull(versionId, "Version id may be null");
+
+        HttpGet getRequest = this.prepareGetRequest(projectId, versionId, GET_ENTITIES_PATH, Lists.mutable.empty());
+        String response = this.execute(getRequest);
+        try
+        {
+            List<DepotEntity> entities = objectMapper.readValue(response, new TypeReference<List<DepotEntity>>()
+            {
+            });
+            return entities.stream().map(entity -> Entity.newEntity(entity.path, entity.classifierPath, entity.content)).collect(Collectors.toList());
```

Comment:
I think you should be able to configure the mapper so that you don't have to do this. Specifically, so that you can just do this:
```suggestion
            return objectMapper.readValue(response, new TypeReference<List<Entity>>()
            {
            });
```